warehouse-management-service/main.go

```diff
@@ -4,45 +4,57 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"warehouse-management-service/internal/config"
+	"warehouse-management-service/internal/handler"
+	"warehouse-management-service/pkg/database/postgres"
 
-	"github.com/gin-gonic/gin"
+	_ "github.com/lib/pq"
+)
+
+const (
+	CONFIG_FILE_PATH_ENV_VARIABLE = "CONFIG_FILE_PATH"
 )
 
 func main() {
-	err := initialize()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "App startup error: %v", err)
-		os.Exit(1)
+	// Get absolute path of config file from env variable
+	configFilePath, ok := os.LookupEnv(CONFIG_FILE_PATH_ENV_VARIABLE)
+	if !ok {
+		fmt.Fprintf(os.Stderr, "%s env variable not set", CONFIG_FILE_PATH_ENV_VARIABLE)
 	}
-}
 
-func initialize() error {
-	err := setupDatabase()
+	// Get config
+	config, err := config.FromFile(configFilePath)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "App startup error: %v", err)
```

Comment:
You can use your logger to output these no?

Reply:
+1

I see there is a logging module already, it could be initialized first and used right after that.

Reply:
Used the logging module here.